nixos/modules/services/misc/gitea.nix

```diff
@@ -303,6 +289,46 @@ in
             };
 
             server = {
+              PROTOCOL = mkOption {
+                type = types.enum [ "http" "https" "fcgi" "http+unix" "fcgi+unix" ];
+                default = "http";
+                description = lib.mdDoc ''Listen protocol. `+unix` means "over unix", not "in addition to."'';
+              };
+
+              HTTP_ADDR = mkOption {
+                type = types.either types.str types.path;
+                default = if lib.hasSuffix "+unix" cfg.settings.server.PROTOCOL then "/run/gitea/gitea.sock" else "0.0.0.0";
+                defaultText = literalExpression ''"0.0.0.0"'';
+                description = lib.mdDoc "Listen address. Must be a path when using a unix socket.";
+              };
+
+              HTTP_PORT = mkOption {
+                type = types.port;
+                default = 3000;
+                description = lib.mdDoc "Listen port. Ignored when using a unix socket.";
+              };
+
+              DOMAIN = mkOption {
+                type = types.str;
+                default = "localhost";
+                description = lib.mdDoc "Domain name of your server.";
+              };
+
+              ROOT_URL = mkOption {
+                type = types.str;
+                default = "http://${cfg.settings.server.DOMAIN}:${toString cfg.settings.server.HTTP_PORT}/";
```

Comment:
How does this handle committing the default port for http and https?

Reply:
I think I meant omitting that the port is not shown in the web ui.

Reply:
It doesn't but it's already an improvement to the previous default that was the constant `http://localhost:3000/`.  
I think handling all cases is too complex to do here, was just trying to make it slightly more useful by default. I can revert it to the constant if you think it's better to not have any attempt to make it dynamic if it's not perfect.

Reply:
Any further comment related to this @SuperSandro2000?  
This is the last unresolved thread.